idaes/models/properties/modular_properties/examples/tests/test_BT_PR_SmoothVLE2.py

```diff
@@ -0,0 +1,891 @@
+                },
+            },
+        },
+    },
+    # Specifying phases
+    "phases": {
+        "Liq": {
+            "type": LiquidPhase,
+            "equation_of_state": Cubic,
+            "equation_of_state_options": {"type": CubicType.PR},
+        },
+        "Vap": {
+            "type": VaporPhase,
+            "equation_of_state": Cubic,
+            "equation_of_state_options": {"type": CubicType.PR},
+        },
+    },
+    # Set base units of measurement
+    "base_units": {
+        "time": pyunits.s,
+        "length": pyunits.m,
+        "mass": pyunits.kg,
+        "amount": pyunits.mol,
+        "temperature": pyunits.K,
+    },
+    # Specifying state definition
+    "state_definition": FTPx,
+    "state_bounds": {
+        "flow_mol": (0, 100, 1000, pyunits.mol / pyunits.s),
+        "temperature": (273.15, 300, 500, pyunits.K),
+        "pressure": (5e4, 1e5, 1e6, pyunits.Pa),
+    },
+    "pressure_ref": (101325, pyunits.Pa),
+    "temperature_ref": (298.15, pyunits.K),
+    # Defining phase equilibria
+    "phases_in_equilibrium": [("Vap", "Liq")],
+    "phase_equilibrium_state": {("Vap", "Liq"): SmoothVLE2},
+    "bubble_dew_method": LogBubbleDew,
+    "parameter_data": {
+        "PR_kappa": {
+            ("benzene", "benzene"): 0.000,
+            ("benzene", "toluene"): 0.000,
+            ("toluene", "benzene"): 0.000,
+            ("toluene", "toluene"): 0.000,
+        }
+    },
+}
+
+
+@pytest.mark.skipif(not cubic_roots_available(), reason="Cubic functions not available")
+class TestBTPR(PropertyTestHarness):
+    def configure(self):
+        self.prop_pack = GenericParameterBlock
+        self.param_args = configuration
+        self.prop_args = {}
+        self.has_density_terms = False
+
+
+# -----------------------------------------------------------------------------
+# Test robustness and some outputs
+@pytest.mark.skipif(not cubic_roots_available(), reason="Cubic functions not available")
+class TestBTExample(object):
+    @pytest.fixture()
+    def m(self):
+        m = ConcreteModel()
+
+        m.fs = FlowsheetBlock(dynamic=False)
+
+        m.fs.props = GenericParameterBlock(**configuration)
+
+        m.fs.state = m.fs.props.build_state_block([1], defined_state=True)
+
+        iscale.calculate_scaling_factors(m.fs.props)
+        iscale.calculate_scaling_factors(m.fs.state[1])
+        return m
+
+    @pytest.mark.integration
+    def test_T_sweep(self, m):
+        assert_units_consistent(m)
+
+        m.fs.obj = Objective(expr=(m.fs.state[1].temperature - 510) ** 2)
+        m.fs.state[1].temperature.setub(600)
+
+        for logP in [9.5, 10, 10.5, 11, 11.5, 12]:
+            m.fs.obj.deactivate()
+
+            m.fs.state[1].flow_mol.fix(100)
+            m.fs.state[1].mole_frac_comp["benzene"].fix(0.5)
+            m.fs.state[1].mole_frac_comp["toluene"].fix(0.5)
+            m.fs.state[1].temperature.fix(300)
+            m.fs.state[1].pressure.fix(10 ** (0.5 * logP))
```

Comment:
What's the point of the 0.5 in the exponent here?

Reply:
To get a half step size, as `range` does not accept non-integer arguments.

Reply:
Then it's not really `logP` anymore, but `2*logP`, which isn't a great variable name. You could get around this issue with `numpy.linspace` or `numpy.logspace`, but that's probably not worth the effort changing.